google/cloud/aiplatform_helpers/value_converter.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from google.protobuf.struct_pb2 import Value
+from proto.marshal.collections.maps import MapComposite
+from proto.marshal import Marshal
+from google.protobuf import json_format
+from google.protobuf.struct_pb2 import Value
+from proto import Message
+from proto.message import MessageMeta
+
+
+def to_value(self: Message) -> Value:
+    """Converts a message type to a :class:`~google.protobuf.struct_pb2.Value` object.
+
+    Args:
+      message: the message to convert
+
+    Returns:
+      the message as a :class:`~google.protobuf.struct_pb2.Value` object
+    """
+    def is_prop(prop):
+        if prop[0].isupper():
+            return False
+        if prop.startswith('_'):
+            return False
+        return True
+
+    props = list(filter(is_prop, dir(self._pb)))
+
+    props_dict = {}
+    for prop in props:
+        props_dict[prop] = getattr(self._pb, prop)
+
+    return json_format.ParseDict(props_dict, Value())
```

Comment:
does this work if some of the values of `props_dict` are nested proto messages?

Reply:
Removed this bit.